### PR TITLE
Client module

### PR DESCRIPTION
**Pull request by xKoem**
Client module which can get user name, put it into queue and get other player name from his queue

### CODE REVIEW

File: Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/queues/QueueConnector.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    private void declareQueue(String queueName) throws IOException {
+        boolean queueAutoDelete = false;
+        boolean queueExclusive = false;
+        boolean queueDurable = false;
+        channel.queueDeclare(queueName, queueDurable, queueExclusive, queueAutoDelete, null);
+    }
+
+    void sendMessage(String queueName, String json) throws IOException {
+        channel.basicPublish("", queueName, null, json.getBytes());
```
Comment (totine) [inline Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/queues/QueueConnector.java]: Pusty string?

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /.idea
-/target
+/*/target
```
Comment (gacikjakub) [inline .gitignore]: ** to have all "target"

File: Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/controllers/ResultController.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import pl.jakubowskiprzemyslaw.tajgertim.models.Player;
+import pl.jakubowskiprzemyslaw.tajgertim.services.QueueService;
+import pl.jakubowskiprzemyslaw.tajgertim.services.SessionService;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+@Controller
+public class ResultController {
```
Comment (gacikjakub) [inline Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/controllers/ResultController.java]: This name of class suggest that is Game Result of some Player

File: Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/services/QueueService.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    public String getOpponentNameFromQueue(String playerQueueName) throws IOException, TimeoutException {
+        Optional<GetResponse> response = Optional.empty();
+        queueConnector.connectToQueue(playerQueueName);
+
+        while (!response.isPresent()) {
+            response = Optional.ofNullable(queueConnector.getResponse(playerQueueName));
+        }
+
+        System.out.println(new String(response.get().getBody(), "UTF-8")); // TODO: 16.07.2018 to be replaced with logger
```
Comment (gacikjakub) [inline Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/services/QueueService.java]: TODO not for MASTER !!!

File: .gitignore (hunk 2 above)
Comment (boczkas) [reply .gitignore]: done

File: Client/pom.xml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            <artifactId>amqp-client</artifactId>
+            <version>5.3.0</version>
+        </dependency>
+    </dependencies>
+
+    <properties>
+        <java.version>1.8</java.version>
+    </properties>
+
+
```
[the remark this answers is not in the document]
Comment (boczkas) [reply Client/pom.xml]: done

File: Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/controllers/PlayerController.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.concurrent.TimeoutException;
+
+@Controller
+public class PlayerController {
+
+    private final PlayerService playerService;
+    private final SessionService sessionService;
+    private final QueueService queueService;
+
+    @Autowired
```
[the remark this answers is not in the document]
Comment (boczkas) [reply Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/controllers/PlayerController.java]: done

File: Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/queues/QueueConnector.java (hunk 1 above)
Comment (boczkas) [reply Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/queues/QueueConnector.java]: Nie został nadany żaden tag kolejki, dlatego tak to wygląda

File: Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/controllers/ResultController.java (hunk 3 above)
Comment (boczkas) [reply Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/controllers/ResultController.java]: Done

File: Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/services/QueueService.java (hunk 4 above)
Comment (boczkas) [reply Client/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/services/QueueService.java]: Done

**Pull request merged**